### PR TITLE
Ensure that top position is not negative

### PR DESCRIPTION
**Pull request by viktorhunko**
Hi @euvl,
I faced with next situation: I have a modal with form inside which works well everywhere except mobile browsers, once some field gets focus and keyboard appears - my modal gets negative top position. I've tried to dig a bit and found out that 'resize' even doesn't fire when keyboard appears, so some calculations use an incorrect max height.
Not sure that solution in this PR is neatly enough, but it should solve the problem.
Thanks,

### CODE REVIEW

File: src/Modal.vue [hunk 1]
```diff
@@ -676,6 +676,7 @@ export default {
 
       this.shift.left -= left - inRange(0, maxLeft, left)
       this.shift.top -= top - inRange(0, maxTop, top)
+      this.shift.top = Math.min(this.shift.top, 0) // to ensure that top is not negative
```
Comment (euvl) [inline src/Modal.vue]: Wait, `Math.min(-100, 0)` will be -100, comment though says "is **not** negative"

Comment (viktorhunko) [reply src/Modal.vue]: dumb typo 🤦‍♂ fixed. should be `Math.max(x, 0)`

**Comment by viktorhunko** [comment]
Dumb typo, it should be `Math.max(x, 0)`, fixed

**Comment by viktorhunko** [comment]
Hi @euvl, recently I've used fork of your repo on my project with changes added by me initially, and I found out that it doesn't solve my issue it all cases - sometimes modal's top position still goes to negative value. I've reviewed it again and found out that it would be better to ensure that `maxTop` is not going to negative value. Fork with latest changes that I've pushed works very well on my project. But I'm not 100% sure that `maxTop` should be always greater or equal to 0, could you confirm that?
Thanks in advance,

**Comment by euvl** [comment]
Hey, I am not 100% sure what is your desired behaviour. Are you trying to "shrink" the modal when it's size is greater than window viewport height?

**Comment by SydorenkoMaksym** [comment]
> Hey, I am not 100% sure what is your desired behaviour. Are you trying to "shrink" the modal when it's size is greater than window viewport height?

Just tried @viktorhunko fix, everything works well while trying to shrink the modal. With negative value of `top` property, top part of modal isn't accessible.

Also I think this could fix #519 issue.

**Comment by viktorhunko** [comment]
> Hey, I am not 100% sure what is your desired behaviour. Are you trying to "shrink" the modal when it's size is greater than window viewport height?

Hey,
the issue: when I have scrollable modal with height greater than the height of viewport sometimes top position goes to negative and you can't scroll to the top of the modal, it is  always behind the viewport (easy to reproduce on mobile phone when some text field in modal get focus and keyboard appears).
desired begavior: don't allow negative value for `top` property for container with `v--modal-box` class, modal will be still scrollable, and most important - fully accessible.
Thanks,

**Comment by euvl** [comment]
Coolio, thanks. Will merge on the weekends 👍

**Comment by kameas** [comment]
@euvl can you please 🙏

**Comment by freearhey** [comment]
@euvl

**Comment by raphaelaleixo** [comment]
Hey!
Anything one can do to help you with this pull request, @euvl ?
Stay safe!

**Comment by euvl** [comment]
thanks guys, much appreciated 👍

**Pull request merged**